### ISSUE TITLE
can't upload with deken.exe

### ISSUE THREAD

**Issue by Spacechild1** [pure-data/deken#204]
'deken package ...' works fine but 'deken upload ...' fails with the message 'no module named keyring' after entering my username and before I can enter the password.

**Comment by umlaeute** [comment]
hmm, the `no module named keyring` is only a warning.

also, i see that you did manage to upload a couple of files at half past two...

**Comment by Spacechild1** [comment]
> hmm, the no module named keyring is only a warning.
I see, but I can't enter my password.
> also, i see that you did manage to upload a couple of files at half past two...
yes, from my Debian machine :-)

**Comment by umlaeute** [comment]
hmm, cannot reproduce. under wine i get:

~~~sh
$ wine deken.exe upload deken-test*.dek
0019:err:winsock:WSAIoctl -> SIO_ADDRESS_LIST_CHANGE request failed with status 0x2733
Deken v0.5.2-22-g5f9da5f
Environment variable DEKEN_USERNAME is not set and the config file C:\users\zmoelnig\.deken\config does not contain a 'username = ...' entry.
To avoid this prompt in future please add a setting to the config or environment.
Please enter username for https://puredata.info:: zmoelnig
No module named 'keyring'
Please enter password for uploading as 'zmoelnig': *****
GPG init failed:
Unable to run gpg (gpg) - it may not be available.
Do you have 'gpg' installed?
- If you've received numerous errors during the initial installation,
  you probably should install 'python-dev', 'libffi-dev' and 'libssl-dev'
  and re-run `deken install`
- On OSX you might want to install the 'GPG Suite'
Signing your package with GPG is optional.
 You can safely ignore this warning if you don't want to sign your package
Your upload was successful.
Please note that it can take up to 24 hours before the package will appear
in deken-searches that allow others to download your package from the
Pd package repository.
No module named 'keyring'
~~~

and the upload succeeds (obviously without gpg-signing the package).
(the "Winsock" error is just some wine-thing)

maybe i changed something recently.

**Comment by Spacechild1** [comment]
OK, I checked again, and I found out that it only fails when I run *deken.exe* in the Msys2 shell:
```
Deken v0.3.0
Could not GPG sign the package.
Environment variable DEKEN_USERNAME is not set and the config file C:\msys64\home\Christof\.deken\config does not contain a 'username = ...' entry.
To avoid this prompt in future please add a setting to the config or environment.
Please enter username for https://puredata.info:: spacechild1
No module named 'keyring'
```
Doesn't continue from there...

However, it works when I run it in *cmd.exe*! Maybe this issue is not worth fixing, but you could add a note in https://github.com/pure-data/deken/blob/master/developer/README.md.

**Comment by umlaeute** [comment]
Am 30. September 2019 17:59:39 MESZ schrieb Christof Ressi <notifications@github.com>:
>OK, I checked again, and I found out that it only fails when I run
>*deken.exe* in the Msys2 shell

interesting. thanks for finding out.

>
>Please enter username for https://puredata.info:: spacechild1
>No module named 'keyring'
>```
>Doesn't continue from there...


as in "returns to the cmdline" or as in "hangs"?

>worth fixing, but you could add a note in
>https://github.com/pure-data/deken/blob/master/developer/README.md.

i'm not sure what to put there.

btw, there's a flag to force asking for a password (-W). does that help?

**Comment by Spacechild1** [comment]
> as in "returns to the cmdline" or as in "hangs"?

"hangs". Another observation: when I force quit with Ctrl+C, the parent *deken.exe* process is closed but the child *deken.exe* process stays around as a zombie.

> i'm not sure what to put there.

Not sure, either. Maybe just a note that it is recommended to run *deken.exe* in *cmd.exe*?

> btw, there's a flag to force asking for a password (-W). does that help?

I guess you meant `-P`. The only difference is that the `No module named 'keyring'` warning doesn't appear. I've tested with both v0.3.0 from the release page and the v0.5.2 nightly build.

**Comment by umlaeute** [comment]
yes, I ment `-P`.

the problem you are describing looks a lot like [this one](https://stackoverflow.com/questions/49858821).

**Comment by umlaeute** [comment]
i think i fixed the issue in ab469ba1d05c177a7fc1f09a93ac57b33ad8d4a5...cc68813fee39fd64221dac70bd6179bc052d5a97

it's supposed to work on both `cmd` and `MSYS2/bash` now.

**Comment by Spacechild1** [comment]
I just tested the CI snapshot - works fine! Thanks a lot!
